src/team/utility/accuracy_metric.py

```diff
@@ -134,6 +136,32 @@ def endpoint_joint_accuracy(rws: RWS, goal_j: np.ndarray) -> float:
     return np.linalg.norm(goal_j - np.array(rws.get_joints_positions()))
 
 
+def gmcc_similarity_metric(reproduced_trajectory: DmpTrajectory,
+                           learned_trajectory: DmpTrajectory) -> float:
+    """
+    Computes the GMCC similarity metric between prediction and ground truth trajectory.
+    GMCC is a measure of how well a linear trasformation between 2 trajectories
+    can be mapped.
+    This similarity metric is invariant to and only to linear trasformations.
+    Paper arxiv url: https://arxiv.org/abs/1906.09802
+
+    :param reproduced_trajectory: trajectory based on task goal requirements
+    :param learned_trajectory:trajectory learned from the demonstrations dataset
+    :return: the GMCC similary metric value between prediction and ground truth
+    trajectory
+    """
+
+    regression = learned_trajectory.joints
+    reproduction = reproduced_trajectory.joints
+    matrix_A = Optimizer(regression, reproduction).find_optimum()
+    prediction = np.matmul(reproduction, matrix_A)
```

Comment:
In section II.A, $A^T$ is used to denote scalar product between two vectors (case with univariate dependent variable $y_t \in \mathbb{R}$), there A is a column vector that just becomes a row vector. In our case, multidimensional trajectories,  H is a matrix.
The steps are:
1. Compute the optimal linear transformation matrix `H` between reproduction (X) and regression (Y) such as XH is as close as possible to Y
2. Calculate prediction $\hat{Y}_{t}$ = XH
3. Calculate GMCC
4. Calculate symmetric GMCC

As long as the matrix multiplication in 1. and 2. are consistent, there is no difference. I used this notation just to avoid transposing trajectories in each and every equation :)